SportsHub/SportsHub.DAL/Data/ApplicationDbContext.cs

```diff
@@ -16,15 +16,19 @@ public ApplicationDbContext(DbContextOptions options) : base(options)
 
         public virtual DbSet<User> Users { get; set; }
         public virtual DbSet<Role> Roles { get; set; }
+        public virtual DbSet<State> States { get; set; }
+        public virtual DbSet<Article> Articles { get; set; }
+        public virtual DbSet<Comment> Comments { get; set; }
+        public virtual DbSet<Category> Categories { get; set; }
 
         protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
         }
 
         protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
-            new UserEntityTypeConfiguration().Configure(modelBuilder.Entity<User>());
-            new RoleEntityTypeConfiguration().Configure(modelBuilder.Entity<Role>());
+            modelBuilder.ApplyConfigurationsFromAssembly(Assembly.GetExecutingAssembly(),
+                x => x.GetInterfaces().Any(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IEntityTypeConfiguration<>)));
```

Comment:
this should be enough modelBuilder.ApplyConfigurationsFromAssembly(GetType().Assembly);

Reply:
Implemented.